crates/starknet/src/transactions.rs

```diff
@@ -154,21 +158,30 @@ impl StarknetTransaction {
     pub fn get_receipt(&self) -> DevnetResult<TransactionReceipt> {
         let transaction_events = self.get_events();
 
+        // decide what units to set for actual fee. V3 transactions are in STRK(FRI)
+        let fee_amount = FeeAmount { amount: self.execution_info.actual_fee };
+        let actual_fee_in_units = match self.inner {
+            Transaction::Declare(DeclareTransaction::Version3(_))
+            | Transaction::DeployAccount(DeployAccountTransaction::Version3(_))
+            | Transaction::Invoke(InvokeTransaction::Version3(_)) => FeeInUnits::STRK(fee_amount),
+            _ => FeeInUnits::WEI(fee_amount),
```

Comment:
Me too, what is your idea

Reply:
I was thinking of logic like:
```
if tx.version() == 3 && tx.type() != L1_HANDLER:
    FRI
else:
    WEI
```

but it would require adding a property/method.

Also, is my pseudocode even correct? I remember seeing similar logic in a dependency of ours, but can't find it anymore

Reply:
In blockifier

Reply:
I see becoming very hard to read with the inclusion of a method, because it will look like that. tx.version() == Felt::from(3)

Reply:
Well for now it's ok, but if there's ever a tx v4, we'll have to refactor this. Alternatively, the transaction itself could return the fee. Resolving.